util/math_helper.h

```diff
@@ -56,6 +56,41 @@ template <typename returnT> struct resultRef {
 
 namespace math {
 
+/* return number of elements in a type */
+int numElements(const float &);
```

Comment:
Nitpicking: what does it mean: "number of elements in a `const float &` type"?
This looks like a dummy function to simplify the code in templates, which examine type traits of composite types like vector, array, struct, etc. Am I right?
If so, I suggest we update the comment to make it less confusing.

Reply:
Updated the comments.